src/hotspot/share/runtime/thread.hpp

```diff
@@ -1125,6 +1125,8 @@ class JavaThread: public Thread {
   void set_requires_cross_modify_fence(bool val) PRODUCT_RETURN NOT_PRODUCT({ _requires_cross_modify_fence = val; })
 
  private:
+  DEBUG_ONLY(void verify_frame_info();)
```

Comment:
If you declare `verify_frame_info` as returning a `bool` (and just put a `return true;` at the end of `JavaThread::verify_frame_info()`), you can call it as:
```
assert(verify_frame_info(), "unexpected frame info");
```

Reply:
You can keep the assert inside `verify_frame_info()` which dumps additional data. (Yes, it's a bit confusing: assert inside an assert :-) )

Reply:
FTR I'm fine with it either way.